collectors/python.d.plugin/freeswitch/freeswitch.chart.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Description: freeswitch netdata python.d module
+# Author: Koldo Aingeru Marcos (badcrc)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+try:
+    import ESL
+    HAS_ESL = True
+except ImportError:
+    HAS_ESL = False
+
+import json
+from random import randint
+
+
+from bases.FrameworkServices.SimpleService import SimpleService
+
+
+DEFAULT_HOST = '127.0.0.1'
+DEFAULT_PORT = 8021
+DEFAULT_PASSWORD = 'ClueCon'
+
+ORDER = [
+    'calls',
+    'registrations',
+    'gateways',
+    'profiles'
+]
+
+
+CHARTS = {
+    'calls': {
+        'options': [None, 'Active Calls', 'calls', 'calls', 'freeswitch.calls', 'line'],
+        'lines': [
+            ['calls','calls','absolute']
+        ]
+    },
+    'registrations': {
+        'options': [None, 'Registrations', 'registrations', 'registrations', 'freeswitch.registrations', 'line'],
+        'lines': [
+            ['registrations','registrations','absolute']
+        ]
+    },
+    'gateways': {
+        'options': [None, 'Gateways', 'gateways', 'gateways', 'freeswitch.gateways', 'line'],
+        'lines': [
+            ['UNREGED','UNREGED','absolute'],
+            ['TRYING','TRYING','absolute'],
+            ['REGISTER','REGISTER','absolute'],
+            ['REGED','REGED','absolute'],
+            ['UNREGISTER','UNREGISTER','absolute'],
+            ['FAILED','FAILED','absolute'],
+            ['FAIL_WAIT','FAIL_WAIT','absolute'],
+            ['EXPIRED','EXPIRED','absolute'],
+            ['NOREG','NOREG','absolute'],
+            ['TIMEOUT','TIMEOUT','absolute'],
+        ]
+    },
+    'profiles': {
+        'options': [None, 'Active Profiles', 'profiles', 'profiles', 'freeswitch.profiles', 'line'],
+        'lines': [
+            ['RUNNING','RUNNING','absolute'],
+            ['DOWN','DOWN','absolute'],
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        self.host = configuration.get('host', DEFAULT_HOST)
+        self.port = configuration.get('port', DEFAULT_PORT)
+        self.password = configuration.get('password', DEFAULT_PASSWORD)
+
+        self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+
+    def _check(self):
+        if not HAS_ESL:
+            self.error("'ESL' package is needed to use freeswitch module, you can install with with pip install python-ESL")
+            return False
+        
+        try:
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+        except:
+            self.error("Cannot connect to FreeSwitch with provided data, check that the service is running")
+            return False
+        
+        return True
+
+    def _get_data(self):
+        if not self.fs.connected():
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
```

Comment:
We created the connection in `check`, how it can be disconnected here? Could you give me a link to the ESL package documentation?

Reply:
I'm trying to find more detailed info than this: https://freeswitch.org/confluence/display/FREESWITCH/Python+ESL But I cannot seem to find it :/

I added this because if freeswitch restarts (changing the configuration, etc...) it dropped the connections and needed to reconnect.

Reply:
The way this is currently structured doesn't appear to guarantee that `self.fs` is connected.

You need to check somehow after recreating the connection that it's actually connected, and should probably return `None` if it's still disconnected.